From da671a49903b977434f51c26df1840c0efd346f6 Mon Sep 17 00:00:00 2001
Message-ID: <da671a49903b977434f51c26df1840c0efd346f6.1740173176.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1740173176.git.sharlatanus@gmail.com>
References: <cover.1740173176.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 17 Feb 2025 16:53:37 +0000
Subject: [PATCH 24/51] gnu: python-halotools: Update to 0.9.3.

* gnu/packages/astronomy.scm (python-halotools): Update to 0.9.3.
[arguments] <phases>: Remove 'build-extensions and use custom 'check.

Change-Id: Ia6ce0c7de0544c0a9ad071aa0ef5dd63ff57c544
---
 gnu/packages/astronomy.scm | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 516969fb3a8..3147674ae89 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -2830,23 +2830,25 @@ (define-public python-gwcs
 (define-public python-halotools
   (package
     (name "python-halotools")
-    (version "0.9.2")
+    (version "0.9.3")
     (source
      (origin
        (method url-fetch)
        (uri (pypi-uri "halotools" version))
        (sha256
-        (base32 "1fa4w8dabr0zx6xwnrdwdipf4s6sg7q25jmq42n9q8n64kjx60vb"))))
+        (base32 "004nqlyiv6gyzmjk840a1hl3j4sgi5xwbfibankwi7281gq4hx3d"))))
     (build-system pyproject-build-system)
     (arguments
      (list
-      ;; Tests are shaky in parallel.
       #:phases
       #~(modify-phases %standard-phases
-          (add-before 'check 'build-extensions
-            (lambda _
-              (setenv "HOME" "/tmp")
-              (invoke "python" "setup.py" "build_ext" "--inplace"))))))
+          ;; Use built library for tests.
+          (replace 'check
+            (lambda* (#:key tests? test-flags #:allow-other-keys)
+              (when tests?
+                (with-directory-excursion #$output
+                  (setenv "HOME" "/tmp")
+                  (apply invoke "pytest" "-vv" test-flags))))))))
     (native-inputs
      (list python-cython-3
            python-extension-helpers
-- 
2.47.1

